From cf35f7af0a3fbab7c799d0a5b160f5a39d8e1f31 Mon Sep 17 00:00:00 2001
From: ZhangPeng <zhangpeng362@huawei.com>
Date: Tue, 24 Dec 2024 16:43:50 +0800
Subject: [PATCH 08/23] Revert "kabi: net: reserve space for xdp subsystem
 related structure"

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/IBDFAV

----------------------------------------------------------------------

This reverts commit f8cb61566576a623971d5cc8dd3cd6229e787e30.

Signed-off-by: ZhangPeng <zhangpeng362@huawei.com>
---
 include/net/xdp.h | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/include/net/xdp.h b/include/net/xdp.h
index 9b9c7dc25eeb..c283668458ca 100644
--- a/include/net/xdp.h
+++ b/include/net/xdp.h
@@ -54,9 +54,6 @@ enum xdp_mem_type {
 struct xdp_mem_info {
 	u32 type; /* enum xdp_mem_type, but known size type */
 	u32 id;
-
-	KABI_RESERVE(1)
-	KABI_RESERVE(2)
 };
 
 struct page_pool;
@@ -77,9 +74,6 @@ struct xdp_rxq_info {
 
 struct xdp_txq_info {
 	struct net_device *dev;
-
-	KABI_RESERVE(1)
-	KABI_RESERVE(2)
 };
 
 enum xdp_buff_flags {
@@ -98,11 +92,6 @@ struct xdp_buff {
 	struct xdp_txq_info *txq;
 	u32 frame_sz; /* frame size to deduce data_hard_end/reserved tailroom*/
 	u32 flags; /* supported values defined in xdp_buff_flags */
-
-	KABI_RESERVE(1)
-	KABI_RESERVE(2)
-	KABI_RESERVE(3)
-	KABI_RESERVE(4)
 };
 
 static __always_inline bool xdp_buff_has_frags(struct xdp_buff *xdp)
@@ -192,11 +181,6 @@ struct xdp_frame {
 	struct net_device *dev_rx; /* used by cpumap */
 	u32 frame_sz;
 	u32 flags; /* supported values defined in xdp_buff_flags */
-
-	KABI_RESERVE(1)
-	KABI_RESERVE(2)
-	KABI_RESERVE(3)
-	KABI_RESERVE(4)
 };
 
 static __always_inline bool xdp_frame_has_frags(struct xdp_frame *frame)
@@ -214,9 +198,6 @@ struct xdp_frame_bulk {
 	int count;
 	void *xa;
 	void *q[XDP_BULK_QUEUE_SIZE];
-
-	KABI_RESERVE(1)
-	KABI_RESERVE(2)
 };
 
 static __always_inline void xdp_frame_bulk_init(struct xdp_frame_bulk *bq)
-- 
2.25.1

